bridges/snowbridge/primitives/core/src/location.rs

```diff
@@ -23,23 +23,30 @@ pub type AgentId = H256;
 
 /// Creates an AgentId from a Location. An AgentId is a unique mapping to an Agent contract on
 /// Ethereum which acts as the sovereign account for the Location.
-/// Resolves Polkadot locations (as seen by Ethereum) to unique `AgentId` identifiers.
-pub type AgentIdOf = HashedDescription<
-	AgentId,
-	(
-		DescribeHere,
-		DescribeFamily<DescribeAllTerminal>,
-		DescribeGlobalPrefix<(DescribeTerminus, DescribeFamily<DescribeTokenTerminal>)>,
-	),
->;
+#[deprecated(note = "Use LocationHashOf instead.")]
+pub type AgentIdOf =
+	HashedDescription<AgentId, (DescribeHere, DescribeFamily<DescribeAllTerminal>)>;
 
 pub type TokenId = H256;
 
 /// Convert a token location (relative to Ethereum) to a stable ID that can be used on the Ethereum
 /// side
+#[deprecated(note = "Use LocationHashOf instead.")]
 pub type TokenIdOf = HashedDescription<
 	TokenId,
-	DescribeGlobalPrefix<(DescribeTerminus, DescribeFamily<DescribeTokenTerminal>)>,
+	DescribeForEthereum<(DescribeTerminus, DescribeFamily<DescribeSnowbridgeTerminal>)>,
+>;
+
+/// Resolves Polkadot locations (as seen by Ethereum) to unique `H256` identifiers.
+/// V1 converter is left here for backwards compatibility
+pub type LocationHashOf = HashedDescription<
```

Comment:
See my comment at https://github.com/vgeddes/polkadot-sdk/pull/28#discussion_r1981776065

Also the documentation for `LocationHashOf` needs to be more detailed, explaining the compatibility issue.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/28/commits/c5bf3d2af59db7cc9f843979ce6aa6a74417a17a